src/lib/core/fiber.h

```diff
@@ -328,6 +328,20 @@ fiber_set_joinable(struct fiber *fiber, bool yesno);
 API_EXPORT int
 fiber_join(struct fiber *f);
 
+/**
+ * Wait until the fiber is dead or timeout exceeded.
+ * Return fiber execution status to the caller or -1
+ * if timeout exceeded and set diag.
+ * The fiber must not be detached (@sa fiber_set_joinable()).
+ * @pre FIBER_IS_JOINABLE flag is set.
```

Comment:
I'd also mention behaviour in case of passing INFINITE timeout

Reply:
fix